go/install/update.go

```diff
@@ -233,16 +256,58 @@ func (u *Updater) Update() (update *keybase1.Update, err error) {
 		return
 	}
 
-	err = u.ApplyUpdate(update)
+	err = u.ApplyUpdate(ui, update)
 	return
 }
 
-func (u *Updater) ApplyUpdate(update *keybase1.Update) (err error) {
+func (u *Updater) PromptForGoAhead(ui libkb.UpdateUI) (err error) {
+
+	u.G().Log.Debug("+ Update.PromptForGoAhead")
+	defer func() {
+		u.G().Log.Debug("- Update.PromptForGoAhead -> %v", err)
+	}()
+
+	if up := u.G().Env.GetUpdatePreferences(); up != nil && up.Auto {
+		u.G().Log.Debug("| going ahead with auto-updates")
+		return nil
+	}
+
+	if ui == nil {
+		return libkb.NoUIError{Which: "UpdateUI"}
+	}
+
+	upa := keybase1.UpdatePromptArg{
+		Version: u.config.Version,
+	}
+	var upr keybase1.UpdatePromptRes
+	upr, err = ui.UpdatePrompt(context.TODO(), upa)
```

Comment:
Can we do a thing where we mutex the updater but don't block. (Cancel existing.)

So for example,
- Prompt for update
- User isn't around for 24 hours
- Updater check happens again
- If updater is in progress, cancel it
- Proceed again